packages/backend-app-api/src/services/implementations/auth/DefaultAuthService.ts

```diff
@@ -82,7 +86,18 @@ export class DefaultAuthService implements AuthService {
 
     const externalResult = await this.externalTokenHandler.verifyToken(token);
     if (externalResult) {
-      return createCredentialsWithServicePrincipal(externalResult.subject);
+      if (
+        externalResult.scope?.pluginIds &&
+        !externalResult.scope.pluginIds.includes(this.pluginId)
+      ) {
+        throw new NotAllowedError('Illegal scope');
+      }
+
+      return createCredentialsWithServicePrincipal(
+        externalResult.subject,
+        undefined,
+        externalResult.scope,
```

Comment:
The scope doesn't get forwarded to plugin request tokens. For example, if you've issued a token that allows a user to call the TechDocs APIs, you will indirectly get access to read from the catalog, or at least enumerate entities. I don't think that's necessarily wrong, but something to document clearly and think a bit more about.

One option could be to forward the external request token, but that kinda goes against what we're trying to do here. We could also forward the scope in the plugin request tokens. That'll keep things more locked down, but it does mean that you'll need to anticipate all of the indirect permissions that you need to allow for a token based on what calls the target plugin is making.

Reply:
Yep I noted the same thing. I don't think we want to deal with transitive permissions though. It makes things much harder to reason about. We have to trust plugins to do the right thing anyway, with requests that come in. The same problem kind of applies to user tokens as well, right? There's a security perspective here for sure but I'd prefer to leave it out of scope here.

Reply:
Yep I think it's the way to go, just gotta make sure it's all clearly communicated too